ISIL-ESTE/Student-Workflow-Organizer
# Implement Sign Up Component with Next.js

## Description

This pull request introduces the implementation of a Sign Up component in Next.js, allowing users to register on our platform. The feature also includes seamless integration with our Node.js backend to securely handle user registration data.

## Changes Made

- Created a new SignUp component in the components folder, enabling users to input their registration details such as name, email, and password.

- Developed the API route for handling user sign-up on the backend. The API route is located in the app/signup directory and is responsible for receiving and processing the data sent from the SignUp component.

- Configured the client-side logic for sending a POST request to the API route with the user's registration data using fetch.

- Implemented server-side validation to ensure proper handling of incoming data, protecting against potential security vulnerabilities with SSR.

## How to Test

- Run the application locally by executing npm run dev and navigate to the SignUp page.

- Enter valid registration details and submit the form.

- Verify that the form data is successfully sent to the Node.js backend.

- Check the server logs to ensure that the registration data is processed correctly.

- Confirm that the user receives proper feedback on the frontend, indicating whether the registration was successful or if there are any errors.
